tests/unit/algos/list/delimiterMaster.spec.js

```diff
@@ -657,6 +677,326 @@ function *generateKeys(count, state) {
                 // ...it should return to skipping by prefix as usual
                 assert.strictEqual(delimiter.skipping(), `${inc(DbPrefixes.Replay)}foo/`);
             });
+
+            it('should not skip over whole prefix when a key equals the prefix and ends with delimiter', () => {
+                for (const prefix of ['prefix/', 'prefix/subprefix/']) {
+                    const delimiter = new DelimiterMaster({
+                        prefix,
+                        delimiter: '/',
+                    }, fakeLogger, vFormat);
+                    for (const testEntry of [
+                        {
+                            key: prefix,
+                            expectedRes: FILTER_ACCEPT,
+                            expectedSkipping: `${prefix}${VID_SEP}`,
+                        },
+                        {
+                            key: `${prefix}${VID_SEP}v1`,
+                            value: '{}',
+                            expectedRes: FILTER_SKIP, // versions get skipped after master
+                            expectedSkipping: `${prefix}${VID_SEP}`,
+                        },
+                        {
+                            key: `${prefix}deleted`,
+                            isDeleteMarker: true,
+                            expectedRes: FILTER_SKIP, // delete markers get skipped
+                            expectedSkipping: `${prefix}${VID_SEP}`,
+                        },
+                        {
+                            key: `${prefix}deleted${VID_SEP}v1`,
+                            isDeleteMarker: true,
+                            expectedRes: FILTER_SKIP,
+                            expectedSkipping: `${prefix}${VID_SEP}`,
+                        },
+                        {
+                            key: `${prefix}deleted${VID_SEP}v2`,
+                            expectedRes: FILTER_SKIP,
+                            expectedSkipping: `${prefix}${VID_SEP}`,
+                        },
+                        {
```

Comment:
Could be useful to add a `notdeleted` key here similarly to the test below, to make sure it is not skipped